app/models/concerns/github_repoable.rb

```diff
@@ -56,9 +56,7 @@ def push_starter_code
     assignment_repository   = GitHubRepository.new(client, github_repo_id)
     starter_code_repository = GitHubRepository.new(client, starter_code_repo_id)
 
-    delete_github_repository_on_failure do
-      assignment_repository.get_starter_code_from(starter_code_repository)
-    end
+    assignment_repository.get_starter_code_from(starter_code_repository)
```

Comment:
This block ditches the use`#delete_github_repository_on_failure` so that it can instead be used in `app/models/group_assignment_repo.rb`